libfreerdp/codec/progressive.c

```diff
@@ -2308,10 +2318,253 @@ INT32 progressive_decompress(PROGRESSIVE_CONTEXT* progressive, const BYTE* pSrcD
 	return rc;
 }
 
+static BOOL progressive_rfx_write_message_progressive_simple(RFX_CONTEXT* context, wStream* s,
+                                                             RFX_MESSAGE* msg)
+{
+	UINT32 blockLen;
+	UINT32 i;
+	UINT32* qv;
+	RFX_TILE* tile;
+	UINT32 tilesDataSize;
+
+	if (context->mode != RLGR1)
+	{
+		WLog_ERR(TAG, "%s: error, RLGR1 mode is required!", __FUNCTION__);
+		return FALSE;
+	}
+
+	/* RFX_PROGRESSIVE_SYNC */
```

Comment:
Both `RFX_PROGRESSIVE_SYNC` and `RFX_PROGRESSIVE_CONTEXT` should only be sent once. This is what the RDS in MS Windows does.

Reply:
Yes, it is a possible improvement, but the current flow is within spec, so I´d prefer fixing the issue at hand first and improve upon that later on.

Reply:
Alright.